go/vt/vttablet/tabletserver/vstreamer/planbuilder.go

```diff
@@ -111,6 +121,42 @@ func (plan *Plan) fields() []*querypb.Field {
 	return fields
 }
 
+// getOpcode returns the equivalent planbuilder opcode for operators that are supported in Filters
+func getOpcode(comparison *sqlparser.ComparisonExpr) (Opcode, error) {
+	var opcode Opcode
+	switch comparison.Operator {
+	case sqlparser.EqualOp:
+		opcode = Equal
+	case sqlparser.LessThanOp:
+		opcode = LessThan
+	case sqlparser.LessEqualOp:
+		opcode = LessThanEqual
+	case sqlparser.GreaterThanOp:
+		opcode = GreaterThan
+	case sqlparser.GreaterEqualOp:
+		opcode = GreaterThanEqual
+	case sqlparser.NotEqualOp:
+		opcode = NotEqual
+	default:
+		return -1, fmt.Errorf("comparison operator %s not supported", comparison.Operator.ToString())
+	}
+	return opcode, nil
+}
+
+// compare returns true after applying the comparison specified in the Filter to the actual data in the column
+func compare(comparison Opcode, columnValue, filterValue sqltypes.Value) (bool, error) {
+	result, err := evalengine.NullsafeCompare(columnValue, filterValue)
+	if err != nil {
+		return false, err
+	}
+	if result == 0 && (comparison == Equal || comparison == LessThanEqual || comparison == GreaterThanEqual) ||
+		result < 0 && (comparison == LessThan || comparison == LessThanEqual || comparison == NotEqual) ||
+		result > 0 && (comparison == GreaterThan || comparison == GreaterThanEqual || comparison == NotEqual) {
+		return true, nil
```

Comment:
Personal preference: break this `if` into 3 `if` statements (`== 0`, `< 0`, `> 0`), each of which ends with `return true, nil`.
Alternatively, a `switch` statement with 3 cases. 
There's just too much `&&` and `||` going on.

Reply:
another alternative is to use bitmasks for the comparisons, so you could write something like:

```
if result == 0  {
  return comparison =| 1 << Equal, nil
}
if result < 0  {
  return comparison =| 1 << LessThan, nil
}
if result > 0  {
  return comparison =| 1 << GreaterThan, nil, nil
}
```

Reply:
I find this syntax confusing. Didn't ever see `=|` used before in `golang`

Reply:
how are you used to checking bitmasks?

Reply:
C-style: `if my_var | 0x0040 { do_something }`

Reply:
sorry for the confusing comment. I was thinking of something like:

```golang
	const (
		NotEqual = 0
		Equal = 1 << iota
		LessThan
		GreaterThan
	)
	var cmp int
	var x sqlparser.ComparisonExpr
	switch x.Operator {
	case sqlparser.EqualOp:
		cmp = Equal
	case sqlparser.LessThanOp:
		cmp = LessThan
	case sqlparser.GreaterThanOp:
		cmp = GreaterThan
	case sqlparser.LessEqualOp:
		cmp = LessThan | Equal
	case sqlparser.GreaterEqualOp:
		cmp = GreaterThan | Equal
	case sqlparser.NotEqualOp:
		cmp = NotEqual
	}
	var result int
	if cmp == NotEqual {
		return result != 0, nil
	}
	if result == 0 {
		return cmp&Equal != 0
	}
	if result < 1 {
		return cmp&LessThan != 0
	}
	if result > 0 {
		return cmp&GreaterThan != 0
	}
```

Reply:
Simplifed the checks to a switch-based one. @shlomi-noach, `newVStreamer()` in vstreamer.go has an explanatory comment on `Filters`.